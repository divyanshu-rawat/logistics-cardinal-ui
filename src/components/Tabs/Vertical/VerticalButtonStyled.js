import styled from 'styled-components';

const VerticalTabButtonStyled = styled('a')`
  height: 44px;
  display: flex;
  flex-flow: row;
  align-items: center;
  position: relative;
  background-color: ${({ theme }) => theme.colors.grey700};
  margin-right: ${({ theme }) => theme.spacings.bit};
  padding: ${({ theme }) => theme.composedSpacings.byte};
  outline-width: 0;
  // it's necessary due to react-bootstrap-table specifity collison
  text-decoration: none !important;
  cursor: pointer;
  text-transform: capitalize;
  &:not(:first-child) {
    margin-top: 2px;
  }
  &:hover {
    text-decoration: none;
  }
  &:active,
  &:hover,
  &:focus {
    background-color: ${({ theme }) => theme.colors.grey800};
  }
  &:before {
    content: '';
    position: absolute;
    left: 0;
    top: 0;
    height: 100%;
    width: 4px;
    opacity: ${({ isActive }) => (isActive ? 1 : 0)};
    background-color: ${({ theme }) => theme.colors.primary100};
    transition: opacity 0.3s;
  }
`;

export default VerticalTabButtonStyled;
