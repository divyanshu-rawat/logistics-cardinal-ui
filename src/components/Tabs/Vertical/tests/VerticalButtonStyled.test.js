import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import VerticalButtonStyled from '../VerticalButtonStyled';
import Theme from '../../../../themes/rooster';

const renderComponent = (isActive = false) => (
  <VerticalButtonStyled theme={Theme} isActive={isActive} />
);

describe('<VerticalButtonStyled />', () => {
  it('should render the component', () => {
    const tree = toJson(shallow(renderComponent()));
    expect(tree).toMatchSnapshot();
  });

  it('should render the component with `isActive` prop as true', () => {
    const tree = toJson(shallow(renderComponent(true)));
    expect(tree).toMatchSnapshot();
  });
});
