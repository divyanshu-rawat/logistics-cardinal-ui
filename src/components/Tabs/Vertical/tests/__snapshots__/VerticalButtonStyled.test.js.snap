// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VerticalButtonStyled /> should render the component 1`] = `
<a
  className="VerticalButtonStyled__VerticalTabButtonStyled-xiMYt iIPbfq"
/>
`;

exports[`<VerticalButtonStyled /> should render the component with \`isActive\` prop as true 1`] = `
<a
  className="VerticalButtonStyled__VerticalTabButtonStyled-xiMYt ewQpgd"
/>
`;
